src/cosmos/CosmosClient.ts

```diff
@@ -1,68 +1,83 @@
+    return await client.cosmos.bank.v1beta1.balance({
+      address,
+      denom,
+    });
   }
 
   async supply(denom: string): Promise<QuerySupplyOfResponse> {
-    const rpc = await this.getRpcClient();
-    const BankQueryClientImpl = cosmos.bank.v1beta1.QueryClientImpl;
-    const queryServiceBank = new BankQueryClientImpl(rpc);
-    const res = await queryServiceBank.supplyOf({ denom });
-    return res;
+    const client = await this.getCosmosRpcClient();
+    return await client.cosmos.bank.v1beta1.supplyOf({
+      denom,
+    });
   }
 
   async validator(validatorAddr: string): Promise<QueryValidatorResponse> {
-    const rpc = await this.getRpcClient();
-    const BankQueryClientImpl = cosmos.staking.v1beta1.QueryClientImpl;
-    const queryServiceBank = new BankQueryClientImpl(rpc);
-    const res = await queryServiceBank.validator({ validatorAddr });
-    return res;
+    const client = await this.getCosmosRpcClient();
+    return await client.cosmos.staking.v1beta1.validator({
+      validatorAddr,
+    });
   }
 
-  async contractInfo(address: string) {
-    const rpc = await this.getRpcClient();
-    const queryService = new QueryClientImpl(rpc);
-    const contract = await queryService.ContractInfo({ address });
-    return contract;
+  async contractInfo(address: string): Promise<QueryContractInfoResponse> {
+    const client = await this.getCosmwasmRpcClient();
+    return await client.cosmwasm.wasm.v1.contractInfo({
+      address,
+    });
   }
 
-  async fetchContractQuery(address: string, queryData: object) {
-    const rpc = await this.getRpcClient();
-    const ContractQueryImpl = cosmwasm.wasm.v1.QueryClientImpl;
-    const contractQuery = new ContractQueryImpl(rpc);
-    const res = await contractQuery.smartContractState({
+  async fetchContractQuery(
+    address: string,
+    queryData: object,
+  ): Promise<QuerySmartContractStateResponse> {
+    const client = await this.getCosmwasmRpcClient();
+    const res = await client.cosmwasm.wasm.v1.smartContractState({
       address,
       queryData: Buffer.from(JSON.stringify(queryData)),
     });
+
     return JSON.parse(Buffer.from(res.data).toString('utf8'));
   }
 
-  async getRpcClient(): Promise<ProtobufRpcClient> {
-    const tmClient = await Tendermint34Client.connect(this.rpcEndpoint);
-    const client = new QueryClient(tmClient);
-    return createProtobufRpcClient(client);
+  async getCosmosRpcClient() {
+    return await cosmos.ClientFactory.createRPCQueryClient({
+      rpcEndpoint: this.rpcEndpoint,
+    });
+  }
+
+  async getOsmosisRpcClient() {
+    return await osmosis.ClientFactory.createRPCQueryClient({
+      rpcEndpoint: this.rpcEndpoint,
+    });
+  }
+
+  async getIbcRpcClient() {
+    return await ibc.ClientFactory.createRPCQueryClient({
+      rpcEndpoint: this.rpcEndpoint,
+    });
+  }
+
+  async getCosmwasmRpcClient() {
+    return await cosmwasm.ClientFactory.createRPCQueryClient({
+      rpcEndpoint: this.rpcEndpoint,
+    });
```

Comment:
Nitpick:
`await` is not needed if the statement is returned directly, also func won't need to have `async` anymore if await is removed from the body